src/asm2wasm.h

```diff
@@ -385,7 +385,7 @@ class Asm2WasmBuilder {
       }
       if (ast[1] == MINUS && ast[2][0] == NUM) {
         double num = -ast[2][1]->getNumber();
-        assert(isInteger32(num));
+        assert(isSInteger32(num));
```

Comment:
this is not right, we can accept *either* a signed or an unsigned value here. Both fit in a double.